src/main/java/edu/cornell/kfs/fp/service/impl/AmazonWebServicesBillingServiceImpl.java

```diff
@@ -0,0 +1,666 @@
+                    if (ObjectUtils.isNotNull(subObjectCode)) {
+                        transactionDTO.setSubObjectCodeNumber(subObjectCode.getFinancialSubObjectCode());
+                    } else {
+                        LOG.info("updateTransactionDTOObjectCodes() Invalid Sub Object Code.");
+                        transactionDTO.setTransactionInputError(true);
+                    }
+                }
+            } else {
+                LOG.info("updateTransactionDTOObjectCodes() Invalid Object Code.");
+                transactionDTO.setTransactionInputError(true);
+            }
+        } else {
+            transactionDTO.setObjectCodeNumber(getTransactionObjectCode());
+        }
+    }
+    
+    private void updateTransactionDTOProjectCodeNumber(AmazonBillingDistributionOfIncomeTransactionDTO transactionDTO, AmazonBillingCostCenterDTO costCenterDTO) {
+        if (StringUtils.isNotBlank(costCenterDTO.getProjectCode())) {
+            ProjectCode projectCode = getProjectCodeService().getByPrimaryId(costCenterDTO.getProjectCode());
+            if (ObjectUtils.isNotNull(projectCode)) {
+                transactionDTO.setProjectCodeNumber(projectCode.getCode());
+            } else {
+                LOG.info("updateTransactionDTOProjectCodeNumber() Invalid Project Code.");
+                transactionDTO.setTransactionInputError(true);
+            }
+        }
+    }
+    
+    private boolean shouldBuildDistributionOfIncomeDocument(AmazonAccountDetail amazonAccountDetail, KualiDecimal cost) {
+        for (DocumentHeader dh : findDocumentHeadersForAmazonDetail(amazonAccountDetail)) {
+            try {
+                DistributionOfIncomeAndExpenseDocument di = (DistributionOfIncomeAndExpenseDocument) getDocumentService().getByDocumentHeaderId(dh.getDocumentNumber());
+                if (di.getTotalDollarAmount().equals(cost)) {
+                    LOG.info("shouldBuildDistributionOfIncomeDocument() DI document number " + di.getDocumentNumber() + " already exists for Amazon account: " + amazonAccountDetail.toString());
+                    return false;
+                }
+            } catch (WorkflowException e) {
+                LOG.error("shouldBuildDistributionOfIncomeDocument() There was a problems creating a DI document from a document header: " + dh.getDocumentNumber(), e); 
+                throw new RuntimeException(e);
+            }
+        }
+        return true;
+    }
+    
+    private Collection<DocumentHeader> findDocumentHeadersForAmazonDetail(AmazonAccountDetail amazonAccountDetail) {
+        Map fieldValues = new HashMap();
+        fieldValues.put(CUKFSConstants.DOCUMENT_HEADER_EXPLANATION_FIELD, buildDocumentExplanation(amazonAccountDetail.getAwsAccount()));
+        fieldValues.put(CUKFSConstants.DOCUMENT_HEADER_DESCRIPTION_FIELD, buildDocumentDescription(amazonAccountDetail.getBusinessPurpose()));
```

Comment:
Does this work? it looks like you are passing in the DB column name rather then the property name? There are constants in KFSPropertConstants.java file in the foundation project for the explanation and doc description fields:
EXPLANATION and DOCUMENT_DESCRIPTION